packages/xstate-wallet/src/store/memory-store.ts

```diff
@@ -0,0 +1,168 @@
+import {Observable, fromEvent} from 'rxjs';
+import {filter} from 'rxjs/operators';
+import {EventEmitter} from 'eventemitter3';
+import * as _ from 'lodash';
+import {getChannelId} from '@statechannels/nitro-protocol';
+
+import {Signature, BigNumber, bigNumberify} from 'ethers/utils';
+type Bytes32 = string;
+type Uint256 = string;
+
+export interface State {
+  channelId: Bytes32;
+  turnNum: BigNumber;
+}
+
+interface DirectFunding {
+  type: 'Direct';
+}
+
+interface IndirectFunding {
+  type: 'Indirect';
+  ledgerId: string;
+}
+
+interface VirtualFunding {
+  type: 'Virtual';
+  jointChannelId: string;
+  guarantorChannelId: string;
+}
+
+interface Guaranteed {
+  type: 'Guarantee';
+  guarantorChannelIds: [string, string];
+}
+
+export type Funding = DirectFunding | IndirectFunding | VirtualFunding | Guaranteed;
+
+export interface Participant {
+  participantId: string;
+  signingAddress: string;
+  destination: string;
+}
+interface SignedState {
+  state: State;
+  signatures: Signature[];
+}
+
+export interface ChannelStoreEntry {
+  states: SignedState[];
+  privateKey: string;
+  participants: Participant[];
+  channelNonce: Uint256;
+  chainId: Uint256;
+  funding?: Funding;
+}
+
+interface CreateAndDirectFund {
+  name: 'CreateAndDirectFund';
+  participants: Participant[];
+}
+interface Message {
+  signedStates?: SignedState[];
+  protocols?: Protocol[];
+}
+
+export type Protocol = CreateAndDirectFund;
+
+export interface StoreV2 {
+  onStateReceived: (channelId: string) => Observable<any>; // observable
```

Comment:
Curious about the names here. If I have a `store: StoreV2`, and I want to be notified about states received for some channel, I would call 
```
store.onStateReceived(channelId).subscribe({
  next(stateReceived) { ... }
})
```

`onStateReceived` sounds like calling `.on` on an event emitter. To me, a more natural thing to call is something like `store.stateReceivedFeed(channelId).subscribe({ ... })` -- a `Feed` is something you can subscribe to.

Reply:
Yeah, that's what I'm currently working with too. This interface currently isn't up-to-date with the latest approach, which is being implemented in the `MemoryStore` below.